### PR TITLE
Fixed a bug that prevented binding an exchange to a queue, in the case where routing keys are not used.

### PR DESCRIPTION
**Pull request by KevinMellott91**
When a queue and exchange are specified, but routing keys are not, then RabbitMQ's `channel.queueBind` method is never called. This prevents the exchange from being bound to the queue, thus messages are not properly routed.

This pull request refactors the logic within `getQueueName` as follows.

* If exchange is NOT specified
  * Declare direct queue
* Otherwise:
  * Declare specified exchange
  * Declare topic queue
  * If routing keys are specified
    * Bind to each routing key
  * Else
    * Bind the queue to the exchange, using an empty routing key (default)

### CODE REVIEW

File: src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala [hunk 1]
```diff
@@ -119,28 +119,38 @@ class RabbitMQReceiver(params: Map[String, String], storageLevel: StorageLevel)
...
-        log.info("declaring topic queue")
-        channel.exchangeDeclare(exchangeName, exchangeType, true)
 
-          val queue = channel.queueDeclare(queueName, true, false, false, new util.HashMap(0)).getQueue()
-
-        for (routingKey: String <- routingKey.split(",")) {
-          log.info("binding to routing key " + routingKey)
-          channel.queueBind(queue, exchangeName, routingKey)
+    // If no exchange was specified, then create a direct connection to the queue.
+    if(exchangeName == null || exchangeName.trim.isEmpty) {
```
Comment (Gschiavon) [inline src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala]: Hi Kevin, in the actual code the exchangeName never will be null. 
I suggest to do it like this:

 private val exchangeName: Option[String] = params.get("exchangeName")

 def getQueueName(channel: Channel): String = {
    // Get the queue name to use (explicit vs auto-generated).
    val queueName = checkQueueName()

    // If no exchange was specified, then create a direct connection to the queue.

      exchangeName match {
        case Some(exchangeName) => {
          log.info(s"declaring exchange '$exchangeName' of type '$exchangeType'")
          channel.exchangeDeclare(exchangeName, exchangeType, true)

          log.info("declaring topic queue")
          channel.queueDeclare(queueName, true, false, false, new util.HashMap(0))

          // Bind the exchange to the queue.
          routingKeys match {
            case Some(routingKey) => {
              // If routing keys were provided, then bind using them.
              for (routingKey: String <- routingKey.split(",")) {
                log.info(s"binding to routing key '$routingKey'")
                channel.queueBind(queueName, exchangeName, routingKey)
              }
            }
        case None =>
              log.info("binding exchange using empty routing key")
              channel.queueBind(queueName, exchangeName, "")
            }
        }
        case None => {
          log.info("exchange was not configured; declaring direct queue")
          channel.queueDeclare(queueName, true, false, false, getParams.asJava)
        }
      }

    queueName
  }

Comment (KevinMellott91) [inline src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala]: Thank you for the suggested change; I have made this update, and all works expected after testing again.

**Comment by KevinMellott91** [comment]
I modified these changes a little more; please see my explanation below and let me know what you think.
* Assume that the exchange is ALWAYS present, since it was previously being assigned the default name of "rabbitmq-exchange" prior to my initial changes.
* Always use the contents of `getParams` when initializing the queue.

**Comment by Gschiavon** [comment]
Thank you Kevin! PR Merged.
Regards.

**Pull request merged**